ReactiveCocoa/DynamicProperty.swift

```diff
@@ -2,65 +2,59 @@ import Foundation
 import ReactiveSwift
 import enum Result.NoError
 
-/// Wraps a `dynamic` property, or one defined in Objective-C, using Key-Value
-/// Coding and Key-Value Observing.
+/// A typed mutable property view to a certain key path of an Objective-C object using
+/// Key-Value Coding and Key-Value Observing.
 ///
-/// Use this class only as a last resort! `MutableProperty` is generally better
-/// unless KVC/KVO is required by the API you're using (for example,
-/// `NSOperation`).
+/// Bindings towards a `DynamicProperty` would be directed to the underlying Objective-C
+/// object, and would not be affected by the deinitialization of the `DynamicProperty`.
 public final class DynamicProperty<Value>: MutablePropertyProtocol {
 	private weak var object: NSObject?
 	private let keyPath: String
+	private let cache: Property<Value>
 
 	/// The current value of the property, as read and written using Key-Value
 	/// Coding.
-	public var value: Value? {
-		get {
-			return object?.value(forKeyPath: keyPath) as! Value
-		}
-
-		set(newValue) {
-			object?.setValue(newValue, forKeyPath: keyPath)
-		}
+	public var value: Value {
+		get { return cache.value }
+		set { object?.setValue(newValue, forKeyPath: keyPath) }
 	}
 
 	/// The lifetime of the property.
 	public var lifetime: Lifetime {
 		return object?.reactive.lifetime ?? .empty
 	}
 
+	/// The binding target of the property.
+	public var bindingTarget: BindingTarget<Value> {
+		return BindingTarget(lifetime: lifetime) { [weak object, keyPath] value in
```

Comment:
This should be able to use the `BindingTarget.init(object:keyPath:)` added below? Or remove that `init`?